### PR TITLE
Updates for First ASP.NET Core App on a Mac Using VS Code

### LINKED ISSUE #1806: Your First ASP.NET Core Application on a Mac Using Visual Studio Code, does not deploy to azure
Looking at this tutorial: [Your First ASP.NET Core Application on a Mac Using Visual Studio Code
](https://docs.asp.net/en/latest/tutorials/your-first-mac-aspnet.html) [[source](https://github.com/aspnet/Docs/blob/master/aspnet/tutorials/your-first-mac-aspnet.rst)]

under [Initialize Azure Website
](https://docs.asp.net/en/latest/tutorials/your-first-mac-aspnet.html#initialize-azure-website)

- Create a new Web App (https://tryappservice.azure.com/) takes me to a page which looks different from the documentation
- link  continuous deployment using Git (https://azure.microsoft.com/en-us/documentation/articles/web-sites-publish-source-control/) returns 404
- Most unfortunately, pushing to the newly created azure site does not appear to deploy anything after running:

<details>
   <summary>Click to expand</summary>

```
C:\Users\jmarolf\OneDrive\Code\aspnetcoreapp (master)
λ git push azure master
Counting objects: 28, done.
Delta compression using up to 8 threads.
Compressing objects: 100% (24/24), done.
Writing objects: 100% (28/28), 40.72 KiB | 0 bytes/s, done.
Total 28 (delta 2), reused 0 (delta 0)
remote: Updating branch 'master'.
remote: Updating submodules.
remote: Preparing deployment for commit id 'b91a1f6722'.
remote: Generating deployment script.
remote: Generating deployment script for Web Site
remote: Generated deployment script files
remote: Running deployment command...
remote: Handling Basic Web Site deployment.
remote: KuduSync.NET from: 'D:\home\site\repository' to: 'D:\home\site\wwwroot'
remote: Copying file: '.gitignore'
remote: Copying file: 'Program.cs'
remote: Copying file: 'project.json'
remote: Copying file: 'project.lock.json'
remote: Copying file: 'Startup.cs'
remote: Copying file: '.vscode\launch.json'
remote: Copying file: '.vscode\tasks.json'
remote: Copying file: 'bin\Debug\netcoreapp1.0\aspnetcoreapp.deps.json'
remote: Copying file: 'bin\Debug\netcoreapp1.0\aspnetcoreapp.dll'
remote: Copying file: 'bin\Debug\netcoreapp1.0\aspnetcoreapp.pdb'
remote: Copying file: 'bin\Debug\netcoreapp1.0\aspnetcoreapp.runtimeconfig.dev.json'
remote: Copying file: 'bin\Debug\netcoreapp1.0\aspnetcoreapp.runtimeconfig.json'
remote: Copying file: 'obj\Debug\netcoreapp1.0\.IncrementalCache'
remote: Copying file: 'obj\Debug\netcoreapp1.0\.SDKVersion'
remote: Copying file: 'obj\Debug\netcoreapp1.0\dotnet-compile-csc.rsp'
remote: Copying file: 'obj\Debug\netcoreapp1.0\dotnet-compile.assemblyinfo.cs'
remote: Copying file: 'obj\Debug\netcoreapp1.0\dotnet-compile.rsp'
remote: Finished successfully.
remote: Running post deployment command(s)...
remote: Deployment successful.
To https://$14242c06-0ee0-4-231-b9ee:sJovw9CRtu0rF3qp1zMMhBMH6mQs7XYnNvwEEJERbvCE5jFvcgsjhpeKtufu@14242c06-0ee0-4-231-b9ee.scm.azurewebsites.net/14242c06-0ee0-4-231-b9ee.git
 * [new branch]      master -> master
```

</details>

I see this: 
image

Not this: 
image

Which I see if I debug locally.

### PR DESCRIPTION
**Pull request by spboyer**
Closes #1806 

@danroth27 @Rick-Anderson

### CODE REVIEW

File: aspnet/tutorials/your-first-mac-aspnet.rst [hunk 1]
```diff
@@ -1,53 +1,88 @@
...
 .. contents:: Sections:
   :local:
   :depth: 1
 
+
 Setting Up Your Development Environment
 ---------------------------------------
 
-To setup your development machine download and install `.NET Core`_ and `Visual Studio Code`_ with the `C# extension <https://marketplace.visualstudio.com/items?itemName=ms-vscode.csharp>`__.
+To setup your development machine download and install `.NET Core`_ and `Visual Studio Code`_ with the `C# extension <https://marketplace.visualstudio.com/items?itemName=ms-vscode.csharp>`__. Node.js and npm is also required, if not already installed visit `nodejs.org <https://nodejs.org/en/download/package-manager/#osx>`_.
```
Comment (danroth27) [inline aspnet/tutorials/your-first-mac-aspnet.rst]: > Node.js and npm is also required, if

Node.js and npm are also required. If

File: aspnet/tutorials/your-first-mac-aspnet.rst [hunk 2]
```diff
@@ -1,53 +1,88 @@
...
 ---------------------------------------
 
-To setup your development machine download and install `.NET Core`_ and `Visual Studio Code`_ with the `C# extension <https://marketplace.visualstudio.com/items?itemName=ms-vscode.csharp>`__.
+To setup your development machine download and install `.NET Core`_ and `Visual Studio Code`_ with the `C# extension <https://marketplace.visualstudio.com/items?itemName=ms-vscode.csharp>`__. Node.js and npm is also required, if not already installed visit `nodejs.org <https://nodejs.org/en/download/package-manager/#osx>`_. 
 
 Scaffolding Applications Using Yeoman
 -------------------------------------
 
-Follow the instruction in :doc:`/client-side/yeoman` to create an ASP.NET Core project.
+We will be using ``yo aspnet`` to generate the **Web Application Basic** template, you may follow the full instructions in :doc:`/client-side/yeoman` to create an ASP.NET Core project which show an **Empty Web** for reference.
```
Comment (danroth27) [inline aspnet/tutorials/your-first-mac-aspnet.rst]: How about just this:

We will use `` :doc:`Yeoman </client-side/yeoman>` `` to generate an initial ASP.NET Core project.

File: aspnet/tutorials/your-first-mac-aspnet.rst [hunk 3]
```diff
@@ -1,53 +1,88 @@
...
+
+Install the necessary yeoman generators and bower (for client side assets) using npm.
```
Comment (danroth27) [inline aspnet/tutorials/your-first-mac-aspnet.rst]: > Install the necessary yeoman generators and bower

Install the ASP.NE TCore generator and also bower

File: aspnet/tutorials/your-first-mac-aspnet.rst [hunk 4]
```diff
@@ -1,53 +1,88 @@
...
+
+.. code-block:: console
+
+    Your project is now created, you can use the following commands to get going
+        cd "MyFirstApp"
+        dotnet restore
+        dotnet build (optional, build will also happen with it's run)
+        dotnet run
+
+.. note:: Mac keyboard symbols - ⌘ : Command, ⌃ : Ctrl, ⇧ : Shift, ⌥ : Option
```
Comment (danroth27) [inline aspnet/tutorials/your-first-mac-aspnet.rst]: Why is this needed? Remove?

File: aspnet/tutorials/your-first-mac-aspnet.rst [hunk 5]
```diff
@@ -1,53 +1,88 @@
...
+    :scale: 100%
+
+Tap "Yes" to add the build and debug assets.
 
-.. image:: your-first-mac-aspnet/_static/dotnet-restore.png
+.. image:: your-first-mac-aspnet/_static/debug-items-added.png 
+    :scale: 100%
 
-You can run commands directly from within Visual Studio Code, including ``dotnet restore`` and any tools referenced in the *project.json* file, as well as custom tasks defined in *.vscode/tasks.json*.
+Tap "Restore" to restore the project dependencies. Alternately, you can enter ``⌘⇧P`` in Visual Studio Code and then type ``dot`` as shown:
```
Comment (danroth27) [inline aspnet/tutorials/your-first-mac-aspnet.rst]: While the Mac UI will use these symbols for indicating keyboard shortcuts, it doesn't look like Apple uses these symbols in their docs: https://support.apple.com/en-us/HT201236

File: aspnet/tutorials/your-first-mac-aspnet.rst [hunk 6]
```diff
@@ -59,39 +94,57 @@ The Search viewlet allows you to quickly search within the folder structure, sea
 
 The Debug viewlet supports interactive debugging of applications.
 
-Finally, Code's editor has a ton of great features. You'll notice unused using statements are underlined and can be removed automatically by using ``command .`` when the lightbulb icon appears. Classes and methods also display how many references there are in the project to them. If you're coming from Visual Studio, Code includes many of the same keyboard shortcuts, such as ``command k c`` to comment a block of code, and ``command k u`` to uncomment.
+Code's editor has a ton of great features. You'll notice unused using statements are underlined and can be removed automatically by using ``⌘ .`` when the lightbulb icon appears. Classes and methods also display how many references there are in the project to them. If you're coming from Visual Studio, Code includes many of the same keyboard shortcuts, such as ``⌘KC`` to comment a block of code, and ``⌘KU`` to uncomment.
+
+More on editor in `Visual Studio Code`_.  
 
-Running Locally Using Kestrel
+Running Locally Using Kestrel
```
Comment (danroth27) [inline aspnet/tutorials/your-first-mac-aspnet.rst]: Remove extra whitespace

File: aspnet/tutorials/your-first-mac-aspnet.rst [hunk 7]
```diff
@@ -59,39 +94,57 @@ The Search viewlet allows you to quickly search within the folder structure, sea
...
+        "type": "build"
+      },
+      "Microsoft.AspNetCore.Server.IISIntegration": "1.0.0",
+      "Microsoft.AspNetCore.Server.Kestrel": "1.0.0",
+      "Microsoft.AspNetCore.StaticFiles": "1.0.0",
+
+
+Using Visual Studio Code Debugger
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+If you chose to have the debug and build assets added to the project
```
Comment (danroth27) [inline aspnet/tutorials/your-first-mac-aspnet.rst]: project:

File: aspnet/tutorials/your-first-mac-aspnet.rst [hunk 8]
```diff
@@ -59,39 +94,57 @@ The Search viewlet allows you to quickly search within the folder structure, sea
...
+
+.. image:: your-first-mac-aspnet/_static/launch-debugger.png
 
+Your default browser will automatically launch and navigate to ``http://localhost:5000``
 
-- Run ``dotnet run`` command to launch the app
+.. image:: your-first-mac-aspnet/_static/myfirstapp.png
 
-- Navigate to ``localhost:5000``:
+- To stop the applicatio, close the browser and hit the "Stop" icon on the debug bar
```
Comment (danroth27) [inline aspnet/tutorials/your-first-mac-aspnet.rst]: applicatio -> application

File: aspnet/tutorials/your-first-mac-aspnet.rst [hunk 9]
```diff
@@ -59,39 +94,57 @@ The Search viewlet allows you to quickly search within the folder structure, sea
...
 
-.. image:: your-first-mac-aspnet/_static/hello-world.png
+.. image:: your-first-mac-aspnet/_static/debugger.png
 
+Using the dotnet commands
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+- Run ``dotnet run`` command to launch the app from terminal/bash
+
+- Navigate to ``http://localhost:5000``
 - To stop the web server enter ``Ctrl+C``.
```
Comment (danroth27) [inline aspnet/tutorials/your-first-mac-aspnet.rst]: Ctrl -> ^ or Control to be consistent with the rest of the doc

**Comment by danroth27** [comment]
Nice update! Just a few minor comments. :watch:

At some point we should add some more content to this tutorial that actually looks at the generated ASP.NET Core project. Right now most of the content talks about Yeoman, Visual Studio Code and Azure. Added https://github.com/aspnet/Docs/issues/1842 to track.

File: aspnet/tutorials/your-first-mac-aspnet.rst (hunk 5 above)
Comment (spboyer) [inline aspnet/tutorials/your-first-mac-aspnet.rst]: @danroth27 that's true, however the documentation on http://code.visualstudio.com uses the symbols.

**Comment by danroth27** [comment]
:shipit:
